### DISCUSSION TITLE
If code-server could run IDEs or apps other than Code would you use them? what apps would you want to see?

### DISCUSSION THREAD

**Discussion by exvuma** [coder/code-server#2143] [Ideas]
For context, I am the PM for the enterprise product and we are considering integrating some of the other editors into code-server for free. This could make it easy to contribute plugins to code-server too. For example, say you wanted to set up a VNC client and server with code-server as well as Code. 

Excited to hear your thoughts! 😄

**Comment by raghavendra-talur** [comment]
I would like to see obsidian.md(https://obsidian.md/) in a hosted form. It is not open source though, I don't know if it can be made part of code-server without source.

**Comment by schmidDan** [comment]
tbh, I'm rarely leaving my browser window since having code-server 😅 if I do so, then mostly for maintenance of my self-hoster code-server (ofc there's Code's terminal, but having a "real" one is still something differenth, though)

Regarding other IDEs: I know some colleagues who would probably look out for [Spyder](https://www.spyder-ide.org/) hosted in a similar format like Code, since it's their go-to IDE.

**Comment by nathanchu** [comment]
I know this would be harder to do, but could it be possible to run any electron app with code server? This seems impossible to me (and probably is), so I'll just give you my reasoning behind it. I code some electron apps, which doesn't really work on code-server, that type of development has to be entirely local. Running an electron app in the browser is definitely possible, as I've seen in code-server, but could you do something reusable? I do know about coder enterprise and that it would cover this exact purpose, but from my understanding, that's just displayed as a canvas, instead of with each individual element. I know this would sort of rival coder enterprise, but coder enterprise still has a dashboard, k8s support, and supports non-electron apps.

---

Otherwise, an IDE that I want is atom, and vim or emacs (so basically a terminal).

**Reply by exvuma** [reply]
While the only IDE possible currently on code-server is VSCode, Electron apps aren't far from being feasible nor tools like obsidian. Appreciate the feedback :)

**Reply by sr229** [reply]
Agee with @exvuma here. Electron applications require a tonload of shims to get it to work on a browser, and to add insult to the injury, the only cloud IDE effort for Atom was from Facebook and was deprecated before it even started.

**Comment by JammSpread** [comment]
I would want to see IntelliJ Idea if Coder doesn't already have that.

**Reply by bpmct** [reply]
Not in code-server yet, but that is definitely something we are looking into. We have this in our enterprise product and will likely open-source it if the interest is there.
